ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveUtils.java

```diff
@@ -334,19 +343,19 @@ public static UnparseTranslator collectUnescapeIdentifierTranslations(ASTNode no
     return unparseTranslator;
   }
 
-  static class ColumnExprProcessor implements SemanticNodeProcessor {
+  static class IdentifierProcessor implements SemanticNodeProcessor {
 
     @Override
     public Object process(Node nd, Stack<Node> stack, NodeProcessorCtx procCtx, Object... nodeOutputs)
-        throws SemanticException {
+            throws SemanticException {
       UnparseTranslator unparseTranslator = ((QuotedIdExpressionContext)procCtx).getUnparseTranslator();
-      ASTNode tokTableOrColNode = (ASTNode) nd;
-      for (int i = 0; i < tokTableOrColNode.getChildCount(); ++i) {
-        ASTNode child = (ASTNode) tokTableOrColNode.getChild(i);
-        if (child.getType() == HiveParser.Identifier) {
-          unparseTranslator.addIdentifierTranslation(child);
-        }
+      ASTNode identifier = (ASTNode) nd;
+      String id = identifier.getText();
+      if (FunctionRegistry.getFunctionInfo(id) != null){
```

Comment:
@kasakrisz, we skip from quoting just the function calls, right?

Reply:
Yes. Hive doesn't support special and whitespace characters in function names.